### PR TITLE
Updated test suite

### PR DESCRIPTION
**Pull request by pkruithof**
Restructured the configuration/extension classes so new configurations are easier to add.

Also tested pretty much every configuration option extensively. And added missing tests for other classes. This made the Behat test suite obsolete, which removes a lot of test-supporting code.

### CODE REVIEW

File: src/TreeHouse/QueueBundle/DependencyInjection/TreeHouseQueueExtension.php [hunk 1]
```diff
@@ -42,11 +46,11 @@ public function load(array $configs, ContainerBuilder $container)
      * @param array            $config
      * @param ContainerBuilder $container
      */
-    protected function loadDriver(array $config, ContainerBuilder $container)
+    private function loadDriver(array $config, ContainerBuilder $container)
     {
         $container->setParameter('tree_house.queue.driver', $config['driver']);
         $container->setAlias(
-            'tree_house.queue.factory',
+            'tree_house.amqp.factory',
```
Comment (cleentfaar) [inline src/TreeHouse/QueueBundle/DependencyInjection/TreeHouseQueueExtension.php]: Shouldn't we try to have every service in this bundle prefixed with `tree_house.queue...`? 

With this format it almost seems like we have a bundle named `TreeHouseAmqpBundle`.

**Comment by cleentfaar** [comment]
very nice! especially like the use of noback's DI tests :+1: !
lgtm

File: src/TreeHouse/QueueBundle/DependencyInjection/TreeHouseQueueExtension.php [hunk 2]
```diff
@@ -31,6 +34,7 @@ public function load(array $configs, ContainerBuilder $container)
         $this->loadConnections($config, $container);
```
Comment (marcojanssen) [inline src/TreeHouse/QueueBundle/DependencyInjection/TreeHouseQueueExtension.php]: Maybe an option to introduce a simple factory pattern to abstract all the 'loading'.

In this case load could consist of several objects:
- Connections
- Publishers
- Exchanges
- Queues

https://github.com/domnikl/DesignPatternsPHP/tree/master/Creational/SimpleFactory

Comment (marcojanssen) [inline src/TreeHouse/QueueBundle/DependencyInjection/TreeHouseQueueExtension.php]: Makes it easier to test as well

File: src/TreeHouse/QueueBundle/DependencyInjection/TreeHouseQueueExtension.php (hunk 1 above)
Comment (pkruithof) [inline src/TreeHouse/QueueBundle/DependencyInjection/TreeHouseQueueExtension.php]: Good point, I'm working on a follow-up PR, I'll change it there right away.

**Comment by marcojanssen** [comment]
Looks great to me

**Pull request merged**